arm-network/2016-09-01/swagger/network.json

```diff
@@ -1672,6 +1672,9 @@
             "description": "Parameters supplied to the create/update NetworkInterface operation"
           },
           {
+            "$ref": "#/parameters/LocationParameter"
```

Comment:
Attempting to perform a request to 'CreateOrUpdatePublicIPAddresses' with the following parameters (which adopt to the spec):
```
{
	"publicIpAddressName": PUBLIC_IP_NAME,
	"resourceGroupName": RESOURCE_GROUP_NAME,
	"subscriptionId": SUBSCRIPTION_ID,
	"properties": {
		"location": "westeurope",
		"publicIPAllocationMethod": "static"
	},
	"api-version": "2016-09-01"
}
```
NOTE: the "properties" vs "parameters" is discussed below. And attempting this request with "parameters" instead of "properties" does return the error mentioned below.

Returns the error:
```
{
	"code": "LocationRequired",
	"message": "The location property is required for this definition."
}
```

With the suggested change, the call does success.

Also, this change does match the call shape defined in the documentation at: 
https://msdn.microsoft.com/en-us/library/azure/mt163590.aspx